src/pandapipes/pf/pipeflow_setup.py

```diff
@@ -561,24 +559,27 @@ def identify_active_nodes_branches(net, hydraulic=True):
                 # set nodes oos that are not connected to any branches with flow > 0 (0.1 is arbitrary
                 # here, any value between 0 and 1 should work, excluding 0 and 1)
                 nodes_connected[fn_tn] = nodes_connected[fn_tn] & (flow > 0.1)
-
+        fn = branch_pit[:, FROM_NODE].astype(np.int32)
+        tn = branch_pit[:, TO_NODE].astype(np.int32)
+        branches_zero = branches_zero_flow(branch_pit)
+        connected_branches_not_zero = branches_connected & ~branches_zero
+        nodes_zero = ~np.isin(
+            np.arange(len(nodes_connected)),
+            np.concatenate([fn[connected_branches_not_zero], tn[connected_branches_not_zero]])
+        ) & nodes_connected
         if get_net_option(net, "transient"):
-            fn = branch_pit[:, FROM_NODE].astype(np.int32)
-            tn = branch_pit[:, TO_NODE].astype(np.int32)
-            branches_zero = branches_zero_flow(branch_pit)
-            connected_branches_not_zero = branches_connected & ~branches_zero
-            nodes_zero = ~np.isin(
-                np.arange(len(nodes_connected)),
-                np.concatenate([fn[connected_branches_not_zero], tn[connected_branches_not_zero]])
-            ) & nodes_connected
             net["_lookups"]["node_zero_flow"] = nodes_zero
             net["_lookups"]["branch_zero_flow"] = branches_zero
+        else:
+            mask_ge = node_pit[:, NODE_TYPE_T] == GE
+            nodes_connected[mask_ge] = ~nodes_zero[mask_ge]
```

Comment:
is there a connectivity check in every bidirectional calculation? I am wondering if we are producing lots of overhead for this. Wouldn't it make more sense to check for zero flow inside the generator component and use that here?

Reply:
I think it would make even more sense to use the definition from before, as stated above in the else-part of the transient check. Any node with zero flow is regarded as disconnected there.

Reply:
Yes, there is a connectivity check in each bidirectional calculation. Actually, I would say that might be correct, as you always need to check in each heat transfer calculation if the mass flow is unequal zero. 

Therefore, I wouldn't do it in the component itself. I do not see the benefit.